### PR TITLE
Fix nmap-ports regex pattern not allowing 80

### PR DESCRIPTION
**Pull request by Donnype**
Allow any decimal with 1-4 digits instead of up to 5000, 500, 30 or 5 (for decimals having the same amount of digits as these numbers)

### Changes

This handles the case where we want to allow arbitrary decimals up to 10.000. The complicated statements are built such that the max is `65535`, but we were not handling this specific case for larger than 5000, 500, 30 or 5. Of course I used 2, 20, 200 and 2000 in my initial unit tests...

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/3639

### QA notes

Please check that you can add the sensible settings for the nmap-ports boefje as described in the issue!

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: boefjes/boefjes/plugins/kat_nmap_ports/schema.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
...
+      "pattern": "^((6553[0-5]|655[0-2]\\d|65[0-4]\\d{2}|6[0-4]\\d{3}|[1-5]\\d{4}|\\d{1,4})|(6553[0-5]|655[0-2]\\d|65[0-4]\\d{2}|6[0-4]\\d{3}|[1-5]\\d{4}|\\d{1,4})-(6553[0-5]|655[0-2]\\d|65[0-4]\\d{2}|6[0-4]\\d{3}|[1-5]\\d{4}|\\d{1,4}))$|^((6553[0-5]|655[0-2]\\d|65[0-4]\\d{2}|6[0-4]\\d{3}|[1-5]\\d{4}|\\d{1,4})|(6553[0-5]|655[0-2]\\d|65[0-4]\\d{2}|6[0-4]\\d{3}|[1-5]\\d{4}|\\d{1,4})-(6553[0-5]|655[0-2]\\d|65[0-4]\\d{2}|6[0-4]\\d{3}|[1-5]\\d{4}|\\d{1,4}))(,((6553[0-5]|655[0-2]\\d|65[0-4]\\d{2}|6[0-4]\\d{3}|[1-5]\\d{4}|\\d{1,4})|(6553[0-5]|655[0-2]\\d|65[0-4]\\d{2}|6[0-4]\\d{3}|[1-5]\\d{4}|\\d{1,4})-(6553[0-5]|655[0-2]\\d|65[0-4]\\d{2}|6[0-4]\\d{3}|[1-5]\\d{4}|\\d{1,4})))+$",
```
Comment (underdarknl) [inline boefjes/boefjes/plugins/kat_nmap_ports/schema.json:9]: Are we sure all the escaped \d (digit) chars are properly De-escaped when we plonk this regex in the HTML pattern attribute?

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Seems to work. Port 80 can be scanned and some other single ports (22, 443) also work. Tried with a range and also with single ports comma separated and a combination of these.

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Comment by underdarknl** [comment]
The form 'should' do frontend validation, and use this regex. However it seems it does not.
https://github.com/minvws/nl-kat-coordination/blob/ebe90dfa67a032a0ef4de27177f6427b706c6a1e/rocky/rocky/templates/forms/json_schema_form.html#L4C1-L4C43

File: boefjes/boefjes/plugins/kat_nmap_ports/schema.json (hunk 1 above)
Comment (Donnype) [reply boefjes/boefjes/plugins/kat_nmap_ports/schema.json:9]: There is no frontend validation as far as I know, but the backends (both Rocky and KATalogus) now work as per the QA feedback

**Comment by Donnype** [comment]
@underdarknl Issue noted, although not my area of expertise

**Review by underdarknl — APPROVED** [review]

**Pull request merged**